conn.go

```diff
@@ -195,6 +196,7 @@ func (conn *Conn) Close() error {
 		close(conn.eavesdropped)
 	}
 	conn.eavesdroppedLck.Unlock()
+	conn.outWait.Wait()
```

Comment:
Closing a go channel does not do any synchronizations with go routines reading from that channels. After we close conn.out there may still be data in that channel to be read/processed and outWorker will still be running. I.e. when conn.Close() returns it may still be possible that outWorker still has some oustanding messages, that may be lost. Please see referenced bug to understand how this problem manifests itself for a user of the library.